packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/MarkersController.java

```diff
@@ -210,20 +219,96 @@ public void onAnimationUpdate(ValueAnimator animation) {
       fadeIn.setInterpolator(fadeInInterpolator);
       fadeIn.start();
     }
-    MarkerController controller = new MarkerController(marker, consumeTapEvents);
+    MarkerController controller = new MarkerController(marker, consumeTapEvents, cozyMarkerData);
     markerIdToController.put(markerId, controller);
     googleMapsMarkerIdToDartMarkerId.put(marker.getId(), markerId);
   }
 
-  private void changeMarker(Object marker) {
-    if (marker == null) {
+  private void changeMarker(Object newMarker) {
+    if (newMarker == null) {
       return;
     }
-    String markerId = getMarkerId(marker);
+    String markerId = getMarkerId(newMarker);
     MarkerController markerController = markerIdToController.get(markerId);
     if (markerController != null) {
-      Convert.interpretMarkerOptions(marker, markerController, cozyMarkerBuilder);
+      final CozyMarkerData startCozyMarkerData = markerController.currentCozyMarkerData;
+      final CozyMarkerData endCozyMarkerData = Convert.toCozyMarkerData(newMarker);
+      final boolean isTheSameMarker = Objects.equals(startCozyMarkerData, endCozyMarkerData);
+
+      if(startCozyMarkerData != null && 
+         endCozyMarkerData != null &&
+         endCozyMarkerData.isAnimated && 
+         !isTheSameMarker){
+
+        animateMarkerTransition(markerController, newMarker, startCozyMarkerData, endCozyMarkerData);
+      }else{
+        Convert.interpretMarkerOptions(newMarker, markerController, cozyMarkerBuilder);
+      }
+    }
+  }
+
+  private void animateMarkerTransition(MarkerController markerController, Object newMarker, CozyMarkerData startMarkerData, CozyMarkerData endMarkerData){
+    final List<BitmapDescriptor> bitmapsForFrame = new ArrayList<BitmapDescriptor>();
+    final Deque<Marker> markersQueue = new ArrayDeque<Marker>();
+
+    final int totalNumberOfSimultaneousMarkers = 3;
+    final int expectedFps = 60;
+    final int framesNumber = (int) (expectedFps * ((float) markersTransitionAnimationDuration / 1000));
+
+    for(int i = 1; i <= framesNumber; i++){
+      final BitmapDescriptor markerIconFrame = BitmapDescriptorFactory.fromBitmap(cozyMarkerBuilder.buildAnimatedMarker(startMarkerData, endMarkerData, (i*1.0f)/(framesNumber*1.0f)));
```

Comment:
slightly refactored version of the code for improved readability
```suggestion
   float progress = (float) i / framesNumber;
BitmapDescriptor markerIconFrame = BitmapDescriptorFactory.fromBitmap(
    cozyMarkerBuilder.buildAnimatedMarker(startMarkerData, endMarkerData, progress)
);
```